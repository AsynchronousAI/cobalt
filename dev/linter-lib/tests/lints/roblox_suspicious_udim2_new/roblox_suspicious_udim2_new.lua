UDim2.new(0)
UDim2.new(0.5)
UDim2.new(1)
UDim2.new(2)
UDim2.new(a)
UDim2.new(1, 1)
UDim2.new(1, 2)
UDim2.new(a, b)
UDim2.new(1, a)
UDim2.new(1, 1, 1)
UDim2.new(a, b, c)
UDim2.new(1, 1, 1, 1)
UDim2.new(a, b, c, d)
UDim2.fromOffset(1, 1)
UDim2.fromScale(1, 1)
