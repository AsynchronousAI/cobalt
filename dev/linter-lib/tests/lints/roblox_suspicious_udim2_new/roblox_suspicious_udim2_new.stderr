error[roblox_suspicious_udim2_new]: UDim2.new takes 4 numbers, but 1 was provided.
  ┌─ roblox_suspicious_udim2_new.lua:1:1
  │
1 │ UDim2.new(0)
  │ ^^^^^^^^^^^^
  │
  = did you mean to use UDim2.fromScale instead?

error[roblox_suspicious_udim2_new]: UDim2.new takes 4 numbers, but 1 was provided.
  ┌─ roblox_suspicious_udim2_new.lua:2:1
  │
2 │ UDim2.new(0.5)
  │ ^^^^^^^^^^^^^^
  │
  = did you mean to use UDim2.fromScale instead?

error[roblox_suspicious_udim2_new]: UDim2.new takes 4 numbers, but 1 was provided.
  ┌─ roblox_suspicious_udim2_new.lua:3:1
  │
3 │ UDim2.new(1)
  │ ^^^^^^^^^^^^
  │
  = did you mean to use UDim2.fromScale instead?

error[roblox_suspicious_udim2_new]: UDim2.new takes 4 numbers, but 1 was provided.
  ┌─ roblox_suspicious_udim2_new.lua:4:1
  │
4 │ UDim2.new(2)
  │ ^^^^^^^^^^^^
  │
  = did you mean to use UDim2.fromOffset instead?

error[roblox_suspicious_udim2_new]: UDim2.new takes 4 numbers, but 1 was provided.
  ┌─ roblox_suspicious_udim2_new.lua:5:1
  │
5 │ UDim2.new(a)
  │ ^^^^^^^^^^^^

error[roblox_suspicious_udim2_new]: UDim2.new takes 4 numbers, but 2 were provided.
  ┌─ roblox_suspicious_udim2_new.lua:6:1
  │
6 │ UDim2.new(1, 1)
  │ ^^^^^^^^^^^^^^^
  │
  = did you mean to use UDim2.fromScale instead?

error[roblox_suspicious_udim2_new]: UDim2.new takes 4 numbers, but 2 were provided.
  ┌─ roblox_suspicious_udim2_new.lua:7:1
  │
7 │ UDim2.new(1, 2)
  │ ^^^^^^^^^^^^^^^
  │
  = did you mean to use UDim2.fromOffset instead?

error[roblox_suspicious_udim2_new]: UDim2.new takes 4 numbers, but 2 were provided.
  ┌─ roblox_suspicious_udim2_new.lua:9:1
  │
9 │ UDim2.new(1, a)
  │ ^^^^^^^^^^^^^^^

error[roblox_suspicious_udim2_new]: UDim2.new takes 4 numbers, but 3 were provided.
   ┌─ roblox_suspicious_udim2_new.lua:10:1
   │
10 │ UDim2.new(1, 1, 1)
   │ ^^^^^^^^^^^^^^^^^^

error[roblox_suspicious_udim2_new]: UDim2.new takes 4 numbers, but 3 were provided.
   ┌─ roblox_suspicious_udim2_new.lua:11:1
   │
11 │ UDim2.new(a, b, c)
   │ ^^^^^^^^^^^^^^^^^^

